main.tf

```diff
@@ -87,6 +99,12 @@ resource "aws_s3_bucket_lifecycle_configuration" "main" {
       }
     }
   }
+  lifecycle {
+    precondition {
+      condition     = !var.enable_lifecycle || length(var.lifecycle_rules) > 0
```

Comment:
Can't we update `count  = var.enable_lifecycle ? 1 : 0` to `count  = length(var.lifecycle_rules) > 0 ? 1 : 0`?
That way we can remove `enable_lifecycle` variable completely?